cuda_next/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+if (NOT DEFINED CCCL_ENABLE_CUDA_NEXT OR CCCL_ENABLE_CUDA_NEXT)
+    cmake_minimum_required(VERSION 3.15)
+
+    project(CUDA_NEXT VERSION 0.1)
+
+    enable_language(CUDA)
+
+    add_library(CUDA_Next INTERFACE)
+    target_link_libraries(CUDA_Next INTERFACE libcudacxx::libcudacxx)
+    target_include_directories(CUDA_Next INTERFACE "${CMAKE_CURRENT_LIST_DIR}/include")
+
+    # TODO add the below to CCCL cmake 
```

Comment:
This todo needs to be done completed before we can merge, so that users of CCCL can use this library.

Reply:
I added some minimal config files in lib/cmake/cuda_next to make it work. There is much more content in these files in other libraries, but I am not sure how much of that we need, at least initially

Reply:
Here are the rest of the changes you need so that consumers of `CCCL` can use CUDA_Next

```
diff --git a/cuda_next/lib/cmake/cuda_next/cuda_next-config.cmake b/cuda_next/lib/cmake/cuda_next/cuda_next-config.cmake
index b0bdb4436..2be26566c 100644
--- a/cuda_next/lib/cmake/cuda_next/cuda_next-config.cmake
+++ b/cuda_next/lib/cmake/cuda_next/cuda_next-config.cmake
@@ -2,8 +2,20 @@ if (TARGET CUDA_Next)
   return()
 endif()
 
+set(CUDA_NEXT_VERSION_MAJOR 0)
+set(CUDA_NEXT_VERSION_MINOR 1)
+set(CUDA_NEXT_VERSION_PATCH 0)
+set(CUDA_NEXT_VERSION_TWEAK 0)
+
+set(CUDA_NEXT_VERSION "${CUDA_NEXT_VERSION_MAJOR}.${CUDA_NEXT_VERSION_MINOR}.${CUDA_NEXT_VERSION_PATCH}.${CUDA_NEXT_VERSION_TWEAK}")
+
 # We don't ship with CTK, so no need for the ugly name (yet)
 add_library(CUDA_Next INTERFACE)
 target_compile_features(CUDA_Next INTERFACE cxx_std_17)
 
 target_include_directories(CUDA_Next INTERFACE ${CMAKE_CURRENT_LIST_DIR}/../../../include)
+
+if (NOT CUDA_Next_CONFIG)
+  set(CUDA_Next_CONFIG "${CMAKE_CURRENT_LIST_FILE}")
+endif()
+find_package_handle_standard_args(CUDA_Next CONFIG_MODE)
diff --git a/lib/cmake/cccl/cccl-config.cmake b/lib/cmake/cccl/cccl-config.cmake
index 882c67938..6c41159ae 100644
--- a/lib/cmake/cccl/cccl-config.cmake
+++ b/lib/cmake/cccl/cccl-config.cmake
@@ -101,7 +101,10 @@ foreach(component IN LISTS components)
         "${cccl_cmake_dir}/../../../cuda_next/lib/cmake/" # Source layout (GitHub)
                                                           # No install layout for now
     )
-    target_link_libraries(CCCL::CCCL INTERFACE CUDA_Next)
+    if (TARGET CUDA_Next AND NOT TARGET CCCL::CUDA_Next)
+      add_library(CCCL::CUDA_Next ALIAS CUDA_Next)
+      target_link_libraries(CCCL::CCCL INTERFACE CUDA_Next)
+    endif()
   else()
     message(FATAL_ERROR "Invalid CCCL component requested: '${component}'")
   endif()
   ```

Reply:
@robertmaynard is there a reason to define the version info in the config file, rather than relying on the config-version file?

Reply:
Also @robertmaynard: Do you know if rapids-cmake will need us to add the header search / custom install dir stuff for this library?

Reply:
> is there a reason to define the version info in the config file, rather than relying on the config-version file?
The scope of variables defined in the version file are local, and not exposed to the caller

> Do you know if rapids-cmake will need us to add the header search / custom install dir stuff for this library?

Not currently as CUDANext says it doesn't support being installed.

Reply:
I've added install rules, but all targets/install rules are opt-in and disabled by default. I've left the header search stuff out for now, we can patch it in if we decide we want it later.